util/progress/progressui/display.go

```diff
@@ -1123,3 +1260,78 @@ func wrapHeight(j []*job, limit int) []*job {
 	}
 	return wrapped
 }
+
+func byteArrayToStringArray(in [][]byte) []string {
+	if len(in) == 0 {
+		return nil
+	}
+
+	out := make([]string, len(in))
+	for i, b := range in {
+		out[i] = string(b)
+	}
+	return out
+}
+
+type logMux struct {
```

Comment:
I think calling this "mux" is confusing as this is just a map. You could argue that the `digestLogs` part does some multiplexing (but more like indexing).

Reply:
I can probably consolidate this with `digestLogs` and rename it. I'm just not the biggest fan of things like this:

```
map[digest.Digest]map[int][]*logEntry
```

This kind of structure is a bit confusing to follow even if it's correct but it's sometimes hard to think of good names for them.